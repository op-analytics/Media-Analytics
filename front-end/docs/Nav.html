<!DOCTYPE html>
<html lang="en">
<head>
    
    <meta charset="utf-8">
    <title>JSDoc: Nav</title>

    <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
    <script src="./build/entry.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link href="https://fonts.googleapis.com/css?family=Muli:100,400,700|Oswald:300|Inconsolata,700" rel="stylesheet">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
    <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
    <link type="text/css" rel="stylesheet" href="styles/app.min.css">
    <link type="text/css" rel="stylesheet" href="styles/iframe.css">
</head>

<body>
    <div id="stickyNavbarOverlay"></div>
    <div class="top-navbar">
        <div class="container">
            <nav class="navbar" role="navigation" aria-label="main navigation">
                <div class="navbar-brand">
                    
                     
                        <h1 class="navbar-item">My React components</h1>
                    
                    <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                    </a>
                </div>
                
            </nav>
        </div>
    </div>
    <div class="container">
        <div class="columns">
            <div class="column is-3" id="sidebarNav">
                <div class="sidebar">
                    <nav>
                        <h2><a href="index.html">Home</a></h2><div class="category"><h3>Components</h3><ul><li><a href="App.html">App</a></li><li><a href="Form.html">Form</a></li><li><a href="Home.html">Home</a></li><li><a href="LineCharts.html">LineCharts</a></li><li><a href="Nav.html">Nav</a></li><li><a href="NotFound.html">NotFound</a></li><li><a href="Timeline.html">Timeline</a></li></ul><h3>Global</h3><ul><li><a href="global.html#cleanDataset">cleanDataset</a></li><li><a href="global.html#createTooltip">createTooltip</a></li></ul></div>
                    </nav>
                </div>
            </div>
            <div class="column is-9-desktop">
                <div class="content" id="main-content-wrapper">
                    <header class="page-title">
                        <p>Components</p>
                        <h1>Nav</h1>
                    </header>
                    
                    




<section>

<header>
    
        <h2>
            &ltNav <i>children</i> <i>title</i> <i>links</i> /&gt
        </h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class='vertical-section'>
                <h3 class="title"><span>Constructor</span></h3>
                <div class="members">
                    <div class="member">


    
    <h4 class="name" id="Nav">
        
        <span class="code-name">
            
                &ltNav <i>children</i> <i>title</i> <i>links</i> /&gt
            
        </span>
    </h4>
    

    
    
    <div class="description">
        <p>Nav A reusable nav bar with a sidebar</p>
    </div>
    







    
        <h5>PropTypes:</h5>
        

<table class="params">
  <thead>
  <tr>
      <th>Name</th>
      <th>Type</th>
      <th>Required</th>
      <th>Description</th>
      <th class="last">Default</th>
  </tr>
  </thead>

  <tbody>
  
        <tr class="deep-level-0">
          <td class="name"><code>children</code></td>
          <td>
            element
          </td>
          <td>No</td>
          <td>The elements to render beside the sidebar.</td>
          <td>null</td>
        </tr>
        
  
        <tr class="deep-level-0">
          <td class="name"><code>title</code></td>
          <td>
            string
          </td>
          <td>No</td>
          <td>The title to render on the app bar.</td>
          <td>''</td>
        </tr>
        
  
        <tr class="deep-level-0">
          <td class="name"><code>links</code></td>
          <td>
            arrayOf
          </td>
          <td>Yes</td>
          <td>The links to render on the sidebar.</td>
          <td></td>
        </tr>
        
  
  </tbody>
</table>
    







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="Shared_Nav.js.html" class="button">View Source</a>
            <span>
                <a href="Shared_Nav.js.html">Shared/Nav.js</a>, <a href="Shared_Nav.js.html#line58">line 58</a>
            </span>
        </p>
    
</dl>





















</div>
                </div>
            </div>
        
    
    </div>
    
        
    
    

    

    

    

    

    

    

    

    

    
</article>

</section>




                </div>
            </div>
        </div>
    </div>

<footer class="footer">
    <div class="content has-text-centered">
        <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.3</a> on Wed Nov 06 2019 19:06:16 GMT+1300 (New Zealand Daylight Time)</p>
        <p class="sidebar-created-by">
            <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
            <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
        </p>
    </div>
</footer>

<script src="scripts/app.min.js"></script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>