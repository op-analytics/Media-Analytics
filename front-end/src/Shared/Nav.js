import AppBar from '@material-ui/core/AppBar';
import CssBaseline from '@material-ui/core/CssBaseline';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import Hidden from '@material-ui/core/Hidden';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MenuIcon from '@material-ui/icons/Menu';
import PropTypes from 'prop-types';
import React from 'react';

const drawerWidth = 240;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0,
    },
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`,
    },
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('sm')]: {
      display: 'none',
    },
  },
  toolbar: theme.mixins.toolbar,
  drawerPaper: {
    width: drawerWidth,
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
}));

function Nav({ children, title }) {
  const classes = useStyles();
  const [mobileOpen, setMobileOpen] = React.useState(false);

  const drawer = (
    <div>
      <div className={classes.toolbar} />
      <Divider />
      <List>
        {['Home', 'Timeline', 'Sentiment Analysis', 'Topic Modeling', 'Latent Association'].map(
          text => (
            <ListItem button key={text}>
              <ListItemText primary={text} />
            </ListItem>
          ),
        )}
      </List>
      <Divider />
      <List>
        {['About', 'Docs', 'Paper'].map(text => (
          <ListItem button key={text}>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
    </div>
  );

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={() => setMobileOpen(!mobileOpen)}
            className={classes.menuButton}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            {title}
          </Typography>
        </Toolbar>
      </AppBar>
      <nav className={classes.drawer}>
        {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
        <Hidden smUp implementation="css">
          <Drawer
            variant="temporary"
            anchor={'right'}
            open={mobileOpen}
            onClose={() => setMobileOpen(!mobileOpen)}
            classes={{
              paper: classes.drawerPaper,
            }}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
          >
            {drawer}
          </Drawer>
        </Hidden>
        <Hidden xsDown implementation="css">
          <Drawer
            classes={{
              paper: classes.drawerPaper,
            }}
            variant="permanent"
            open
          >
            {drawer}
          </Drawer>
        </Hidden>
      </nav>
      <main className={classes.content}>
        <div className={classes.toolbar} />
        {children}
      </main>
    </div>
  );
}

Nav.defaultProps = {
  children: null,
  title: '',
};

Nav.propTypes = {
  children: PropTypes.element,
  title: PropTypes.string,
};

export default Nav;
