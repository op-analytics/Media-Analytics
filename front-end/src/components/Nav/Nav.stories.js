import React from 'react';
import { StoreProvider, createStore, action } from 'easy-peasy';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '.';

export default {
  title: 'Components/Navbar',
  component: Nav,
};

const getWrapper = authenticated => {
  const initialState = { user: { authenticated, logout: action(() => ({})) } };
  const store = createStore(initialState);

  // eslint-disable-next-line react/display-name
  return storyFn => (
    <StoreProvider store={store}>
      <Router>{storyFn()}</Router>
    </StoreProvider>
  );
};

const authenticated = getWrapper(true);
const unauthenticated = getWrapper(false);

const links = [
  [
    { href: '/', text: 'Home' },
    { href: '/timeline', text: 'Timeline' },
    { href: '/sentiment-analysis', text: 'Sentiment Analysis' },
    { href: '/topic-modeling', text: 'Topic Modeling' },
    { href: '/latent-association', text: 'Latent Association' },
  ],
];

export const NavbarNoDivider = () =>
  unauthenticated(() => <Nav title="navbar" links={links} />);

export const NavbarWithDivider = () =>
  unauthenticated(() => (
    <Nav
      title="navbar-divider"
      links={[
        [
          { href: '/', text: 'Home' },
          { href: '/timeline', text: 'Timeline' },
          { href: '/sentiment-analysis', text: 'Sentiment Analysis' },
          { href: '/topic-modeling', text: 'Topic Modeling' },
          { href: '/latent-association', text: 'Latent Association' },
        ],
        [
          { href: '/about', text: 'About' },
          { href: '/docs', text: 'Docs' },
          { href: '/paper', text: 'Paper' },
        ],
      ]}
    />
  ));

export const NavbarAuthenticated = () =>
  authenticated(() => <Nav title="navbar" links={links} />);
