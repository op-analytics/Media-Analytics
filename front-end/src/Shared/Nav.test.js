import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

const getWrapper = () => shallow(<Nav />);

describe('Nav', () => {
  it('Should render', () => {
    const nav = getWrapper();
    expect(nav.exists('nav')).toEqual(true);
  });

  it('Should render correctly', () => {
    const nav = getWrapper();
    expect(nav).toMatchSnapshot();
  });
});
