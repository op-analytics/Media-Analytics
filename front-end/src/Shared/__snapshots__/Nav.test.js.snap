// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should render correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-3"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        noWrap={true}
        variant="h6"
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <nav
    className="makeStyles-drawer-2"
  >
    <Hidden
      implementation="css"
      smUp={true}
    >
      <WithStyles(ForwardRef(Drawer))
        ModalProps={
          Object {
            "keepMounted": true,
          }
        }
        anchor="right"
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-6",
          }
        }
        onClose={[Function]}
        open={false}
        variant="temporary"
      >
        <div>
          <div
            className="makeStyles-toolbar-5"
          />
          <WithStyles(ForwardRef(Divider)) />
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Home"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Home"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Timeline"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Timeline"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Sentiment Analysis"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Sentiment Analysis"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Topic Modeling"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Topic Modeling"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Latent Association"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Latent Association"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
          <WithStyles(ForwardRef(Divider)) />
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="About"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="About"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Docs"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Docs"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Paper"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Paper"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
    <Hidden
      implementation="css"
      xsDown={true}
    >
      <WithStyles(ForwardRef(Drawer))
        classes={
          Object {
            "paper": "makeStyles-drawerPaper-6",
          }
        }
        open={true}
        variant="permanent"
      >
        <div>
          <div
            className="makeStyles-toolbar-5"
          />
          <WithStyles(ForwardRef(Divider)) />
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Home"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Home"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Timeline"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Timeline"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Sentiment Analysis"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Sentiment Analysis"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Topic Modeling"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Topic Modeling"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Latent Association"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Latent Association"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
          <WithStyles(ForwardRef(Divider)) />
          <WithStyles(ForwardRef(List))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="About"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="About"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Docs"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Docs"
              />
            </WithStyles(ForwardRef(ListItem))>
            <WithStyles(ForwardRef(ListItem))
              button={true}
              key="Paper"
            >
              <WithStyles(ForwardRef(ListItemText))
                primary="Paper"
              />
            </WithStyles(ForwardRef(ListItem))>
          </WithStyles(ForwardRef(List))>
        </div>
      </WithStyles(ForwardRef(Drawer))>
    </Hidden>
  </nav>
  <main
    className="makeStyles-content-7"
  >
    <div
      className="makeStyles-toolbar-5"
    />
  </main>
</div>
`;
